pipeline_dp/report_generator.py

```diff
@@ -22,26 +26,32 @@ class ReportGenerator:
   and generates a report.
   """
 
-    def __init__(self, params):
+    def __init__(self,
+                 params,
+                 method_name: str,
+                 is_public_partition: Optional[bool] = None):
         """Initialize the ReportGenerator."""
         self._params_str = None
         if params:
-            self._params_str = str(params)
+            self._params_str = agg.parameters_to_readable_string(
+                params, is_public_partition)
+        self._method_name = method_name
         self._stages = []
 
-    def add_stage(self, text: str):
+    def add_stage(self, stage_description: Union[Callable, str]):
```

Comment:
Why can the description be a Callable? (maybe we can add that to the doc)

Reply:
added, PTAL